controllers/provisioning_controller.go

```diff
@@ -398,7 +398,7 @@ func apiServerHost() (string, error) {
 	}
 	host := config.Host
 	if strings.Contains(apiServerURL.Host, ":") {
-		host = strings.Split(apiServerURL.Host, ":")[0]
+		host, _, _ = net.SplitHostPort(apiServerURL.Host)
```

Comment:
What do you think about checking the error? It's probably relatively safe that url.Parse gave us back good data but just in case...

```suggestion
		host, _, err = net.SplitHostPort(apiServerURL.Host)
		if(err != nil) {
			return "", err
		}
```

Reply:
Good idea to check anyway. An ill formed API server address is unlikely but as you suggested, it is probably a good idea to check.